pipeline/merger/src/main/scala/uk/ac/wellcome/platform/merger/rules/ItemsRule.scala

```diff
@@ -63,96 +62,60 @@ object ItemsRule extends FieldMergeRule with MergerLogging {
   }
 
   /**
-    * If there is 1 Sierra item, we replace the `PhysicalLocation` with
-    * the one from the Calm record.
-    *
-    * Otherwise we add the item to the items list.
+    * Miro: we merge the location into the Sierra work with a single item
+    * as we assume that this is the only item that the Miro image
+    * could be associated with.
     *
-    * This logic is going to be removed soon as we make the Calm
-    * record the `target`.
+    * METS: As there is only 1 Sierra work, we assume that the METS work
+    * iss associated with that and merge the locations onto the Sierra item.
     */
-  private val mergeCalmItems = new PartialRule {
-    val isDefinedForTarget: WorkPredicate = WorkPredicates.sierraWork
-    val isDefinedForSource: WorkPredicate = WorkPredicates.calmWork
+  private val mergeIntoSingleItemSierraTarget = new PartialRule {
+    val isDefinedForTarget: WorkPredicate = WorkPredicates.singleItemSierra
+    val isDefinedForSource
+      : WorkPredicate = WorkPredicates.metsWork or WorkPredicates.miroWork
 
     def rule(target: UnidentifiedWork,
              sources: NonEmptyList[TransformedBaseWork]): FieldData = {
 
-      // The Calm transformer always creates a single item with a physical
-      // location so this is safe
-      val calmItem = sources.head.data.items.head
-      val calmLocation = calmItem.locations.head
-
-      target.data.items match {
-        case List(sierraItem) =>
-          List(
-            sierraItem.copy(
-              // We remove any `PhysicalLocation`s and add the calmLocation
-              // as the only `PhysicalLocation`
-              locations = calmLocation :: sierraItem.locations.filter(_ match {
-                case _: PhysicalLocation => false
-                case _                   => true
-              })
-            )
-          )
-        case multipleItems => calmItem :: multipleItems
-      }
-    }
-  }
+      // This is safe due to the `singleItemSierra` predicate
+      val sierraItem = target.data.items.head
 
-  /**
-    * If there is 1 Sierra item, we add the location from the METS record
-    * else we just append METS items as we wouldn't know which item to
-    * augment.
-    */
-  private val mergeMetsItems = new PartialRule {
-    val isDefinedForTarget: WorkPredicate = WorkPredicates.sierraWork
-    val isDefinedForSource: WorkPredicate = WorkPredicates.singleItemDigitalMets
+      val metsSources = sources.filter(WorkPredicates.metsWork)
```

Comment:
This filtering is redundant as you're partitioning the list of sources, performing the ssame transforms to each partition, and then joining them; you could simply do:
```
val sierraItem = target.data.items.head
val sourceLocations = sources.flatMap(_.data.items).flatMap(_.locations)
List(
  sierraItem.copy(
    locations = sierraItem.locations ++ sourceLocations
  )
)
```

Reply:
Could do, but it feels like simplifying the code at the expense of expressing business logic, which I have spent quite a long time gathering - from that perspective, does it feel like it's worth keeping?

Reply:
I think the business logic is captured in `isDefinedForSource`, the code is just redundant rather than simplifiably complex. You could call `sourceLocations` in my suggestion (eg) `metsAndMiroLocations` to be super explicit?

Reply:
imo the issue with having things like this to express business logic is that they are actually confusing to read - I spent a little while reading it making sure that I had understood correctly that it was redundant, assuming that it was doing kind of filtering or transformation on top of what the rule does

Reply:
🤷 I found it easier to say "Merge the mets and miro locations onto the sierra item". Non fussed either way. It's been changed to have less redundancy.